API: Add Path.addCircle, addRoundRect, close

lessp opened the pull request:
This adds bindings for `Path.` `addCircle`, `addRoundRect`, `close`. This is new ground for me, so please make sure I've not missed something! 😄 

https://api.skia.org/classSkPath.html#a2daa24dc5e277a487c80e2de4469bfb3

lessp committed:
API: Add addCircle

lessp committed:
API: Path.addRoundRect, Path.close

lessp committed:
Formatting

Et7f3 commented on src/wrapped/bindings/SkiaWrappedBindings.re:
```diff
@@ -478,7 +478,21 @@ module M = (F: FOREIGN) => {
 
     let allocate = foreign("sk_path_new", void @-> returning(t));
     let delete = foreign("sk_path_delete", t @-> returning(void));
+    float;
```
?

lessp replied:
Must have gotten there by accident! Thanks!

Et7f3 reviewed (commented):
Why a image is added but unused ?

lessp commented on src/wrapped/bindings/SkiaWrappedBindings.re:
```diff
@@ -478,7 +478,21 @@ module M = (F: FOREIGN) => {
 
     let allocate = foreign("sk_path_new", void @-> returning(t));
     let delete = foreign("sk_path_delete", t @-> returning(void));
+    float;
```
```suggestion
```

lessp committed:
Remove magic float

lessp committed:
Chore: Remove example-image

lessp commented:
> Why a image is added but unused ?

Removed it and added it to `.gitignore`, thanks!

lessp commented:
> > Why a image is added but unused ?
> 
> Removed it and added it to `.gitignore`, thanks!

Anything else holding this back @Et7f3?

Et7f3 commented on src/Skia.re:
```diff
@@ -459,6 +459,9 @@ module RRect = {
 };
 
 module Path = {
+  type pathDirection = SkiaWrapped.Path.pathDirection;
+  let pathDirection = SkiaWrapped.Path.pathDirection;
+
```
```suggestion
```

File "src/Skia.re", line 462, characters 2-53:
Error (warning 34): unused type pathDirection.
File "src/Skia.re", line 463, characters 6-19:
Error (warning 32): unused value pathDirection.

Et7f3 committed:
Remove unused value/type pathDirection

File "src/Skia.re", line 462, characters 2-53:
Error (warning 34): unused type pathDirection.
File "src/Skia.re", line 463, characters 6-19:
Error (warning 32): unused value pathDirection.

Et7f3 commented on example/TestApp.re:
```diff
@@ -34,6 +34,13 @@ let draw = canvas => {
   Path.lineTo(path, 590., 50.);
   Path.cubicTo(path, -490., 50., 1130., 430., 50., 430.);
   Path.lineTo(path, 590., 430.);
+  Path.addCircle(path, 100., 300., 100., Clockwise);
+
+  let roundRect = Rect.makeLtrb(300., 400., 240., 280.);
+  Path.addRoundRect(path, roundRect, 25., 25., Clockwise);
+  Path.addCircle(path, 100., 300., 100., Clockwise);
+
+  Path.close(path);
```
```suggestion
  Path.close(path);
  Path.addCircle(path, 100., 300., 100., Clockwise);

  let roundRect = Rect.makeLtrb(300., 400., 240., 280.);
  Path.addRoundRect(path, roundRect, 25., 25., Clockwise);
  Path.addCircle(path, 100., 300., 100., Clockwise);
```

For me I don't see the close take effect expect I move before other call. Is it a bug only on my computer ? And you have two call with the same coordinates is it intended ? It make the stoke a bit more thick.

Et7f3 replied:
I think it is ok:
`close() has no effect if SkPath is empty or last SkPath SkPath::Verb is kClose_Verb.` addCircle should add a closed shape.

Et7f3 commented:
Overwise seem good to merge

Et7f3 commented:
I see in the man that their is a default value to ClockWise. Maybe when you re export you could define a function with a default arg. :) Otherwise seem good to merge.

lessp committed:
Update example/TestApp.re

Co-Authored-By: Et7f3 <cadeaudeelie@gmail.com>

lessp committed:
API: Make path-direction an optional, name radius

Et7f3 commented on example/TestApp.re:
```diff
@@ -34,6 +34,12 @@ let draw = canvas => {
   Path.cubicTo(path, -490., 50., 1130., 430., 50., 430.);
   Path.lineTo(path, 590., 430.);
+  Path.close(path);
+  Path.addCircle(path, 100., 300., 100., ());
```
```suggestion
  Path.addCircle(path, 100., 200., 75., ());
```
So we can see both circle :)

Et7f3 committed:
Move circle so we can see both

lessp commented:
Perfect, thanks @Et7f3!
